File: backend/moyeo/src/main/java/com/mo/moyeo/domain/notification/entity/Notification.java
package com.mo.moyeo.domain.notification.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;
import java.util.Map;

@Getter
@Entity
@Builder
@AllArgsConstructor
@Table(name = "notifications")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "notification_id")
    private Long id;

    @Column(name = "user_id", nullable = false)
    private Long userId;

    @Column(name = "transaction_id", nullable = false)
    private Long transactionId;

    @Column(name = "title", nullable = false, length = 100)
    private String title;

    @Column(name = "body", nullable = false)
    private String body;

    @Column(name = "data")
    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, String> data;

    @Column(name = "received_at")
    private LocalDateTime receivedAt;
}

File: frontend/app/src/main/java/com/d108/moyeo/domain/model/Notification.kt
package com.d108.moyeo.domain.model

data class Notification(
    val transactionId: Long,        // 거래 목록 id
    val title: String,              // 거래 제목

    // 거래 내용
    val time: String? = null,       // 거래 시각
    val sender: String? = null,     // 보낸 사람
    val amount: String? = null,     // 거래량
    val balance: String? = null,    // 거래 후 잔액

    val boxId: Long? = null,        // data.box 있으면 Long 으로
    val receivedAt: String,         // 날짜 문자열
)